presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/AbstractContentSchemaReader.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.schema;
+
+import io.prestosql.plugin.kafka.KafkaTableHandle;
+
+import java.util.Optional;
+
+public abstract class AbstractContentSchemaReader
+        implements ContentSchemaReader
+{
+    @Override
+    public Optional<String> readKeyContentSchema(KafkaTableHandle tableHandle)
+    {
+        return readSchema(tableHandle.getKeyDataSchemaLocation(), tableHandle.getKeySubject());
```

Comment:
What if we could pass the `KafkaTableHandle` to the underlying implementation ?

Reply:
It would need to differentiate between key and message, so this class abstracts that so subclasses just pass the table handle. Does that sound good to you?